### PR TITLE
Fix relative symlink support in TarFile

### LINKED ISSUE #77303: TarFile fails to extract even when symbolic link references into the output directory
I already have an open issue about TarFile not wanting to create symbolic links that point outside the output directory: https://github.com/dotnet/runtime/issues/74140. 

Triggered by https://github.com/dotnet/runtime/issues/77096, I was curious to see if `podman` would have the same issue as `docker`.

I did:
```
$ podman pull hello-world
$ podman save hello-world -o hello-world.tar # uncompressed when saving a local image
```

The resulting archive fails to extract:
```
Unhandled exception. System.IO.IOException: Extracting the Tar entry '../8e128b102ca8161cb297e3aa73240b8e537af67422fd4c22bf43c296c91d13ad.tar' would have resulted in a link target outside the specified destination directory: '/tmp/bca4a72dc384466ab6abd59cc585bd5a'
   at System.Formats.Tar.TarEntry.GetDestinationAndLinkPaths(String destinationDirectoryPath)
   at System.Formats.Tar.TarEntry.ExtractRelativeToDirectoryAsync(String destinationDirectoryPath, Boolean overwrite, SortedDictionary`2 pendingModes, CancellationToken cancellationToken)
   at System.Formats.Tar.TarFile.ExtractToDirectoryInternalAsync(Stream source, String destinationDirectoryPath, Boolean overwriteFiles, Boolean leaveOpen, CancellationToken cancellationToken)
   at System.Formats.Tar.TarFile.ExtractToDirectoryInternalAsync(Stream source, String destinationDirectoryPath, Boolean overwriteFiles, Boolean leaveOpen, CancellationToken cancellationToken)
   at Program.<Main>$(String[] args) in /tmp/console/Program.cs:line 6
   at Program.<Main>(String[] args)
```

The symbolic link it considers problematic is actually still pointing inside the output directory.
These are the archive entries:
```
8e128b102ca8161cb297e3aa73240b8e537af67422fd4c22bf43c296c91d13ad.tar(RegularFile) -> 
c0e8b50899a56dcd91634bfbb7a8336f2c0f6f6a9bbb5ad9b747dc1547aaa9f4.json(RegularFile) -> 
00335d90e4f471aae0fdbae54765dd7a5a1cb276c55c2fc75bf64ab20c5c97d1/layer.tar(SymbolicLink) -> ../8e128b102ca8161cb297e3aa73240b8e537af67422fd4c22bf43c296c91d13ad.tar
00335d90e4f471aae0fdbae54765dd7a5a1cb276c55c2fc75bf64ab20c5c97d1/VERSION(RegularFile) -> 
00335d90e4f471aae0fdbae54765dd7a5a1cb276c55c2fc75bf64ab20c5c97d1/json(RegularFile) -> 
manifest.json(RegularFile) -> 
repositories(RegularFile) -> 
```

cc @carlossanlop @Jozkee @am11

### PR DESCRIPTION
**Pull request by am11**
Fixes https://github.com/dotnet/runtime/issues/77303

Depends on https://github.com/dotnet/runtime-assets/pull/280

### CODE REVIEW

**Comment by am11** [comment]
@tmds, this gets us closer to tar utilities.

File attributes: I noticed with GNU and BSD `tar(1)`, the file  `layer.tar` in extraction dir has the following permissions: `rwxrwxrwx` and date is of unix epoch's (Jan  1  1970), while we have `rwxr-xr-x` and current date.

**Comment by am11** [comment]
@tmds, this gets us closer to tar utilities.

File attributes: I noticed with GNU and BSD `tar(1)`, the file  `layer.tar` in extraction dir has these permissions: `rwxrwxrwx` and date is of unix epoch's (Jan  1  1970), while we assign `rwxr-xr-x` and current date.

File: src/libraries/System.Formats.Tar/tests/TarFile/TarFile.ExtractToDirectoryAsync.Stream.Tests.cs [hunk 1]
```diff
@@ -110,6 +110,29 @@ public async Task ExtractEntry_DockerImageTarWithFileTypeInDirectoriesInMode_Suc
             }
         }
 
+        [Fact]
+        public async Task ExtractEntry_PodmanImageTarWithRelativeSymlinksPointingInExtractDirectory_SuccessfullyExtracts_Async()
```
Comment (tmds) [inline src/libraries/System.Formats.Tar/tests/TarFile/TarFile.ExtractToDirectoryAsync.Stream.Tests.cs:114]: This has nothing to do with `podman`, so I would not include that in the test name.

**Comment by tmds** [comment]
If we're changing this code, I think it would be good to make a decision about https://github.com/dotnet/runtime/issues/74140.
I think the behavior is restrictive without a good reason.

File: src/libraries/System.Formats.Tar/tests/TarFile/TarFile.ExtractToDirectoryAsync.Stream.Tests.cs (hunk 1 above)
Comment (am11) [reply src/libraries/System.Formats.Tar/tests/TarFile/TarFile.ExtractToDirectoryAsync.Stream.Tests.cs:114]: The other test before this one has nothing to do with docker either. Name makes it just easier to spot from logs.

Comment (tmds) [reply src/libraries/System.Formats.Tar/tests/TarFile/TarFile.ExtractToDirectoryAsync.Stream.Tests.cs:114]: We can even build this file as part of a test, it's a valid one. (Unlike the one created by `docker`).

**Comment by tmds** [comment]
> File attributes: I noticed with GNU and BSD tar(1), the file layer.tar in extraction dir has these permissions: rwxrwxrwx and date is of unix epoch's (Jan 1 1970), while we assign rwxr-xr-x and current date.

Can you create an issue about this?

File: src/libraries/System.Formats.Tar/tests/TarFile/TarFile.ExtractToDirectoryAsync.Stream.Tests.cs (hunk 1 above)
Comment (tmds) [reply src/libraries/System.Formats.Tar/tests/TarFile/TarFile.ExtractToDirectoryAsync.Stream.Tests.cs:114]: `Podman` in this name is just noise imo.

**Comment by am11** [comment]
>  I think the behavior is restrictive without a good reason.

There are two things which throw different IOExceptions:
* file being extracted has destination outside the extraction directory. Throws: `IOException(TarExtractingResultsFileOutside)`
* symlink is pointing to something outside the extraction directory (e.g. /usr/bin/foo). Throws: `IOException(TarExtractingResultsLinkOutside)`

the security concerns mentioned in https://github.com/dotnet/runtime/issues/74140 apply to the first one, but the second one seems harmless given the target of symlink are normally not validated upon symlink creation (it can point to a non-existing path).

**Comment by tmds** [comment]
> the security concerns mentioned in https://github.com/dotnet/runtime/issues/74140 apply to the first one, but the second one seems harmless given the target of symlink are normally not validated upon symlink creation (it can point to a non-existing path).

Yes, https://github.com/dotnet/runtime/issues/74140 is about the behavior being to restrictive in the second case.

@Jozkee @carlossanlop @jeffhandley @adamsitnik can you make a decision about the desired behavior?

**Comment by am11** [comment]
@adamsitnik, currently, System.Formats.Tar allows symlink target to be inside the archive and disallows if the target is pointing to path outside the archive. So far, this PR hasn't changed the behavior. The bug appears when the symlink target is pointing to the path inside the archive with _relative_ paths, the symlink validation (`TarExtractingResultsLinkOutside`) fails. In addition to fixing the bug, this PR is also restoring the original symlink path post-validation, so we don't turn a relative target into absolute one in symlink's target after extracting the archive (matches GNU and BSD tar utilities).

Allowing the target of symlink to path outside the archive is a separate discussion (#74140, awaiting decision) and it can be fixed in a separate PR.

**Comment by tmds** [comment]
> Allowing the target of symlink to path outside the archive is a separate discussion (https://github.com/dotnet/runtime/issues/74140, awaiting decision) and it can be fixed in a separate PR.

Rather than changing this code once more, review it once more, and maybe backport it once more too, why don't we make a decision?

**Comment by am11** [comment]
Hmm, because they are separate issues: https://github.com/dotnet/runtime/issues/77303#issuecomment-1286661730..

**Comment by tmds** [comment]
Yes, and still very related. https://github.com/dotnet/runtime/issues/77303 is a bug in the code written for https://github.com/dotnet/runtime/issues/74140.

**Comment by am11** [comment]
Right, and this is a bug fix with accompanying tests. If decision to allow link target outside the extraction directory will take longer (https://github.com/dotnet/runtime/issues/74140 is awaiting decision for some time now), we should fix this bug regardless. The code change for 74140 is straightforward, so I am fine to fix it here as well. Lets see what the area owners think about next steps.

**Comment by am11** [comment]
FWIW, it will take this change to allow target of symlinks (and hardlinks) outside the archive: https://github.com/am11/runtime/commit/fae7cd29191401c964b8fb46f0f1b6376ec665b9 (remove validation of link target, update two existing tests and add two additional tests for non-existing paths).

**Comment by tmds** [comment]
> FWIW, it will take this change to allow target of symlinks (and hardlinks) outside the archive: https://github.com/am11/runtime/commit/fae7cd29191401c964b8fb46f0f1b6376ec665b9 (remove validation of link target, update two existing tests and add two additional tests for non-existing paths).

You must also add the symlink check (see https://github.com/dotnet/runtime/issues/74140) to prevent 'using' a symlink to place files outside the output directory.

**Comment by am11** [comment]
> You must also add the symlink check (see https://github.com/dotnet/runtime/issues/74140) to prevent 'using' a symlink to place files outside the output directory.

I am not sure how to produce that tar file. Do you have an test case handy which isn't failing with my change, while it should?

**Comment by tmds** [comment]
This is about introducing a new check that prevents an 'evil' tar file to write files outside the output directory. You should be able to create the tar file using the .NET API.

For example:
```
link: SymbolicLink to /tmp
link/file: RegularFile
``` 
Extracting this file without the check, would lead to writing to `/tmp/file` independent of the output directory.

**Comment by am11** [comment]
> You should be able to create the tar file using the .NET API.

How exactly?

**Comment by am11** [comment]
Have you actually found that kind of tar file before? I am not sure how do you think it is possible to construct that tar file using the .NET API (or any other producer for that matter). That's why I asked for the test repro (or the tar file) you are talking about.

**Comment by tmds** [comment]
This is a specially crafted file meant to trigger a security problem. You can't create it from file system. The `TarWriter` API allows you to create it because it will write whatever entries you ask it to.

**Comment by am11** [comment]
Ok. Since you have ideas what needs to be done for https://github.com/dotnet/runtime/issues/74140, I am not pursuing it. As it stands, this PR is only fixing the bug reported in https://github.com/dotnet/runtime/issues/77303.

@Jozkee, can you take a look?

**Comment by Jozkee** [comment]
> Rather than changing this code once more, review it once more, and maybe backport it once more too, why don't we make a decision?

Between https://github.com/dotnet/runtime/issues/77303 and https://github.com/dotnet/runtime/issues/74140, is one of the issues worth backporting while the other is not? 

Without the answer I think keeping them as separate PRs would allow us to nicely cherry-pick in case we choose to backport just one.

File: src/libraries/System.Formats.Tar/tests/TarFile/TarFile.CreateFromDirectory.File.Tests.cs [hunk 2]
```diff
@@ -225,6 +225,45 @@ public void SkipRecursionIntoDirectorySymlinks()
             Assert.Null(reader.GetNextEntry()); // file.txt should not be found
         }
 
+        [ConditionalFact(typeof(MountHelper), nameof(MountHelper.CanCreateSymbolicLinks))]
+        public void WriteSymlinkTargetsInsideTheArchiveRoundtrips()
```
Comment (Jozkee) [inline src/libraries/System.Formats.Tar/tests/TarFile/TarFile.CreateFromDirectory.File.Tests.cs:229]: nit:

I assume you didn't add this to [`./tests/TarWriter/TarWriter.WriteEntry.Entry.Roundtrip.Tests.cs`](https://github.com/dotnet/runtime/blob/main/src/libraries/System.Formats.Tar/tests/TarWriter/TarWriter.WriteEntry.Entry.Roundtrip.Tests.cs) because there are only `TarWriter` tests there. 

Could you please create a similar file in this folder, say `TarFile.CreateFromDirectory.File.Roundtrip.cs`?

**Comment by am11** [comment]
> is one of the issues worth backporting while the other is not?

IMO, this one is worth backporting since we currently allow symlinks within the extraction directory. The validation logic is wrong when the symlink is created with relative path; the bug this PR is fixing. It is not uncommon, e.g. in https://github.com/libunwind/libunwind/archive/refs/tags/v1.6.2.tar.gz, README is a relative symlink to README.md.

> I think keeping them as separate PRs would allow us to nicely cherry-pick

I agree. Keeping bugfixes separate from changing the (intentional) behavior is pretty normal; if not recommended.

File: src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarEntry.cs [hunk 3]
```diff
@@ -349,11 +349,17 @@ internal Task ExtractRelativeToDirectoryAsync(string destinationDirectoryPath, b
                     throw new InvalidDataException(SR.TarEntryHardLinkOrSymlinkLinkNameEmpty);
                 }
 
-                linkTargetPath = GetSanitizedFullPath(destinationDirectoryPath, LinkName);
+                linkTargetPath = GetSanitizedFullPath(destinationDirectoryPath,
+                    Path.IsPathFullyQualified(LinkName) ? LinkName : Path.Join(Path.GetDirectoryName(fileDestinationPath), LinkName));
```
Comment (Jozkee) [inline src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarEntry.cs:353]: Is it possible to move this inside `GetSanitizedFullPath`?

File: src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarEntry.cs [hunk 4]
```diff
@@ -349,11 +349,17 @@ internal Task ExtractRelativeToDirectoryAsync(string destinationDirectoryPath, b
...
+
                 if (linkTargetPath == null)
                 {
                     throw new IOException(SR.Format(SR.TarExtractingResultsLinkOutside, LinkName, destinationDirectoryPath));
                 }
+
+                // after TarExtractingResultsLinkOutside validation, preserve the original
+                // symlink target path (to match behavior of other utilities).
+                linkTargetPath = LinkName;
```
Comment (Jozkee) [inline src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarEntry.cs:362]: How are other symlink cases able to preserve the correct link target (relative or absolute)?

File: src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarEntry.cs (hunk 3 above)
Comment (am11) [reply src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarEntry.cs:353]: `GetSanitizedFullPath` does not have the 'is it a link' context.

File: src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarEntry.cs (hunk 4 above)
Comment (am11) [reply src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarEntry.cs:362]: AFAICT, this code is used by all symlink normalizations.

File: src/libraries/System.Formats.Tar/tests/TarFile/TarFile.CreateFromDirectory.File.Tests.cs (hunk 2 above)
Comment (am11) [reply src/libraries/System.Formats.Tar/tests/TarFile/TarFile.CreateFromDirectory.File.Tests.cs:229]: Sure, I added it next to the existing test (we have only a few (two) existing tests related to symlinks).

File: src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarEntry.cs (hunk 4 above)
Comment (Jozkee) [reply src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarEntry.cs:362]: Ah, so it appears that prior to your PR, we were always extracting symlinks with an absolute target. As part of the outside-of-directory-bug, links with a relative target would be created with an incorrect target, even inside-of-extracted-directory. 

if `LinkName` is `./file,txt`, the symlink will be created pointing to `destinationDirectory/file.txt`, regardless of where the link was in the directory tree. This change is also fixing that.

File: src/libraries/System.Formats.Tar/tests/TarFile/TarFile.CreateFromDirectory.File.Roundtrip.cs [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+
+using System.IO;
+using Xunit;
+
+namespace System.Formats.Tar.Tests
+{
+    public class TarFile_CreateFromDirectory_Roundtrip_Tests : TarTestsBase
+    {
+        [ConditionalFact(typeof(MountHelper), nameof(MountHelper.CanCreateSymbolicLinks))]
+        public void WriteSymlinkTargetsInsideTheArchiveRoundtrips()
```
Comment (Jozkee) [inline src/libraries/System.Formats.Tar/tests/TarFile/TarFile.CreateFromDirectory.File.Roundtrip.cs:12]: Could you please extend the test to have a variety of link target paths and the link itself being in multiple levels (i.e: at 0 (root), and at 1).
e.g:
```
link-> ./file.txt // pass.
link-> ../file.txt // should throw TarExtractingResultsLinkOutside
subDir1/link-> ./file.txt // pass.
subDir/link-> ../file.txt // pass.
subDir/link-> ../../file.txt // should throw TarExtractingResultsLinkOutside
```

File: src/libraries/System.Formats.Tar/tests/TarFile/TarFile.CreateFromDirectory.File.Roundtrip.cs [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+            FileInfo? fileInfo = new(symlinkPath);
+            Assert.Equal(symlinkTargetPath, fileInfo.LinkTarget);
+
+            FileSystemInfo? symlinkTarget = File.ResolveLinkTarget(symlinkPath, returnFinalTarget: true);
+            Assert.True(File.Exists(symlinkTarget.FullName));
+        }
+
+        [ConditionalTheory(typeof(MountHelper), nameof(MountHelper.CanCreateSymbolicLinks))]
+        [InlineData("../file.txt", null)]
+        [InlineData("../../file.txt", "subDirectory")]
```
Comment (tmds) [inline src/libraries/System.Formats.Tar/tests/TarFile/TarFile.CreateFromDirectory.File.Roundtrip.cs:53]: Can you add a test case with an absolute path?

File: src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarEntry.cs (hunk 4 above)
Comment (tmds) [reply src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarEntry.cs:362]: Probably `GetDestinationAndLinkPaths` doesn't even need to return a path for the link target, as we can directly use `LinkName` when creating the link.

File: src/libraries/System.Formats.Tar/tests/TarFile/TarFile.CreateFromDirectory.File.Roundtrip.cs [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+{
+    public class TarFile_CreateFromDirectory_Roundtrip_Tests : TarTestsBase
+    {
+        [ConditionalTheory(typeof(MountHelper), nameof(MountHelper.CanCreateSymbolicLinks))]
+        [InlineData("subDirectory/file.txt", "./file.txt", "subDirectory")]
+        [InlineData("./file.txt", "../file.txt", "subDirectory")]
+        [InlineData("subDirectory1/file.txt", "../file.txt", "subDirectory1/subDirectory1.1")]
+        [InlineData("subDirectory1/subDirectory1.1/file.txt", "./file.txt", "subDirectory1/subDirectory1.1")]
+        [InlineData("./file.txt", "./file.txt", null)]
+        public void SymlinkRelativeTargets_InsideTheArchive_RoundtripsSuccessfully(string targetFilePath, string symlinkTargetPath, string subDirectory)
```
Comment (tmds) [inline src/libraries/System.Formats.Tar/tests/TarFile/TarFile.CreateFromDirectory.File.Roundtrip.cs:17]: Could the first argument of this test be omitted and replaced by `Path.Combine(subDirectory, symlinkTargetPath)`?

File: src/libraries/System.Formats.Tar/tests/TarFile/TarFile.CreateFromDirectory.File.Roundtrip.cs [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+            string sourceDirectoryName = Path.Join(root.Path, "baseDirectory");
+            Directory.CreateDirectory(sourceDirectoryName);
+
+            string destinationDirectoryName = Path.Join(root.Path, "destinationDirectory");
+            Directory.CreateDirectory(destinationDirectoryName);
+
+            string sourceSubDirectory = Path.Join(sourceDirectoryName, subDirectory);
+            if(subDirectory != null)  Directory.CreateDirectory(sourceSubDirectory);
+
+            File.Create(Path.Join(sourceDirectoryName, targetFilePath)).Dispose();
```
Comment (tmds) [inline src/libraries/System.Formats.Tar/tests/TarFile/TarFile.CreateFromDirectory.File.Roundtrip.cs:32]: Do we need to create this file?

File: src/libraries/System.Formats.Tar/tests/TarFile/TarFile.CreateFromDirectory.File.Roundtrip.cs (hunk 6 above)
Comment (am11) [reply src/libraries/System.Formats.Tar/tests/TarFile/TarFile.CreateFromDirectory.File.Roundtrip.cs:53]: Absolute path tests already exist `Extract_LinkEntry_TargetOutsideDirectory`.

File: src/libraries/System.Formats.Tar/tests/TarFile/TarFile.CreateFromDirectory.File.Roundtrip.cs (hunk 8 above)
Comment (am11) [reply src/libraries/System.Formats.Tar/tests/TarFile/TarFile.CreateFromDirectory.File.Roundtrip.cs:32]: Yes. the last assert is checking if target file (resolved symlink path) exists, i.e. all files are accounted for.

File: src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarEntry.cs (hunk 4 above)
Comment (am11) [reply src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarEntry.cs:362]: We need to call `GetDestinationAndLinkPaths` for validation.

Comment (tmds) [reply src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarEntry.cs:362]: I mean:
```
private (string, string?) GetDestinationAndLinkPaths(string destinationDirectoryPath)
```
can probably changed to:
```
private string GetDestinationAndLinkPaths(string destinationDirectoryPath)
```
as we shouldn't use the second item of the tuple (and can directly use `LinkName` instead).

Comment (am11) [reply src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarEntry.cs:362]: Then we will be checking 'is it a link' condition again at callsite (we already check it here for validation).

File: src/libraries/System.Formats.Tar/tests/System.Formats.Tar.Tests.csproj [hunk 9]
```diff
@@ -2,7 +2,7 @@
   <PropertyGroup>
     <TargetFrameworks>$(NetCoreAppCurrent)-windows;$(NetCoreAppCurrent)-Unix</TargetFrameworks>
     <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
-    <StringResourcesPath>$(LibrariesProjectRoot)/Common/tests/Resources/Strings.resx</StringResourcesPath>
+    <StringResourcesPath>$(MSBuildProjectDirectory)\..\src\Resources\Strings.resx</StringResourcesPath>
```
Comment (Jozkee) [inline src/libraries/System.Formats.Tar/tests/System.Formats.Tar.Tests.csproj:5]: is this change intended?

Comment (am11) [reply src/libraries/System.Formats.Tar/tests/System.Formats.Tar.Tests.csproj:5]: Yes, for SR.TarExtractingResultsLinkOutside.

Comment (Jozkee) [reply src/libraries/System.Formats.Tar/tests/System.Formats.Tar.Tests.csproj:5]: No other test project uses `MSBuildProjectDirectory`. Can't you just use a relative path? 
```suggestion
    <StringResourcesPath>..\src\Resources\Strings.resx</StringResourcesPath>
```

That's what we do in other projects.

File: src/libraries/System.Formats.Tar/tests/TarFile/TarFile.CreateFromDirectory.File.Roundtrip.cs [hunk 10]
```diff
@@ -0,0 +1,79 @@
...
+
+            string sourceSubDirectory = Path.Join(sourceDirectoryName, subDirectory);
+            if(subDirectory != null)  Directory.CreateDirectory(sourceSubDirectory);
+
+            File.CreateSymbolicLink(Path.Join(sourceSubDirectory, "linkToFile"), symlinkTargetPath);
+
+            TarFile.CreateFromDirectory(sourceDirectoryName, destinationArchive, includeBaseDirectory: false);
+
+            using FileStream archiveStream = File.OpenRead(destinationArchive);
+            Exception exception = Record.Exception(() => TarFile.ExtractToDirectory(archiveStream, destinationDirectoryName, overwriteFiles: true));
```
Comment (Jozkee) [inline src/libraries/System.Formats.Tar/tests/TarFile/TarFile.CreateFromDirectory.File.Roundtrip.cs:74]:
```suggestion
            Exception exception = Assert.Throws<IOException>(() => TarFile.ExtractToDirectory(archiveStream, destinationDirectoryName, overwriteFiles: true));
```

File: src/libraries/System.Formats.Tar/tests/System.Formats.Tar.Tests.csproj (hunk 9 above)
Comment (am11) [reply src/libraries/System.Formats.Tar/tests/System.Formats.Tar.Tests.csproj:5]: > Can't you just use 

You are asking a different question now. Not sure why it is bothering you this much, it is 100% same thing!

Comment (Jozkee) [reply src/libraries/System.Formats.Tar/tests/System.Formats.Tar.Tests.csproj:5]: I didn't know it was the same thing. In that case, is fine.

**Review by jozkee — APPROVED** [review]

**Comment by Jozkee** [comment]
Thanks.

**Comment by am11** [comment]
Thanks for the reviews @tmds and @Jozkee.
